hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ReplicationManager.java

```diff
@@ -186,16 +197,18 @@ private void run() {
       while (running) {
         try {
           final long start = Time.monotonicNow();
-          final List<ContainerID> containerIds =
+          final Set<ContainerID> containerIds =
               containerManager.getContainerIDs();
-          containerIds.parallelStream().forEach(this::processContainer);
-          LOG.debug("Replication Monitor Thread took {} milliseconds for" +
+          containerIds.forEach(this::processContainer);
+          LOG.info("Replication Monitor Thread took {} milliseconds for" +
                   " processing {} containers.", Time.monotonicNow() - start,
               containerIds.size());
-          Thread.sleep(interval);
+          if (!Thread.interrupted()) {
+            Thread.sleep(interval);
+          }
         } catch (InterruptedException ex) {
           // Wakeup and process the containers.
-          LOG.debug("Replication Monitor Thread got interrupt exception.");
+          LOG.debug("Replication Monitor Thread got interrupted.");
```

Comment:
Sorry I missed this earlier. We should call Thread.getCurrentThread().interrupt() here. https://stackoverflow.com/a/4906814

Reply:
Addressed by using Object.wait instead of Thread.sleep